lib/fs.js

```diff
@@ -2095,6 +2096,7 @@ function WriteStream(path, options) {
   if (!(this instanceof WriteStream))
     return new WriteStream(path, options);
 
+  path = getPathFromURL(path, true);
```

Comment:
If we add it here then `this.path = getPathFromURL(path);` below can just be `this.path = path`, also `path` is ignored if `options.fd !== undefined` so maybe

```js
if (options.fd !== undefined) {
  this.path = getPathFromURL(path);
}
```

Same about `ReadStream`

Reply:
`validatePath` only accepts `Uint8Array` this will work is Stream path is always `Uint8Array` or `string`.

Reply:
@timotew The path in the fs stream classes can be falsy if `options.fd` is provdided, e.g. `fs.createReadStream(null, { fd: fd, encoding: 'utf8' })` is valid. From what I can tell, `validatePath` accepts both `string` and `Uint8Array`.